soc/st/stm32/stm32u5x/poweroff.c

```diff
@@ -12,13 +12,44 @@
 
 #include <stm32_ll_cortex.h>
 #include <stm32_ll_pwr.h>
+#include <stm32_ll_usart.h>
+#include <stm32_ll_dma.h>
+
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32u5_dma)
+#define DT_DRV_COMPAT st_stm32_uart
+#define TURN_OFF_UART_DMA(inst)									\
+	{											\
+		USART_TypeDef *usart = (USART_TypeDef *)DT_INST_REG_ADDR(inst);			\
+		if (DT_INST_DMAS_HAS_NAME(inst, tx)) {						\
+			while (LL_DMA_IsEnabledChannel(						\
+				(DMA_TypeDef *)DT_INST_DMAS_CTLR_BY_NAME(inst, tx),		\
+				 DT_INST_DMAS_CHANNEL_BY_NAME(inst, tx))) {			\
+					/* wait for DMA to be disabled */			\
+			}									\
+			LL_USART_DisableDMAReq_TX(usart);					\
+		}										\
+		if (DT_INST_DMAS_HAS_NAME(inst, rx)) {						\
+			while (LL_DMA_IsEnabledChannel(						\
+				(DMA_TypeDef *)DT_INST_DMAS_CTLR_BY_NAME(inst, rx),		\
+				DT_INST_DMAS_CHANNEL_BY_NAME(inst, rx))) {			\
+					/* wait for DMA to be disabled */			\
+			}									\
+			LL_USART_DisableDMAReq_RX(usart);					\
+		}										\
+	}
+#endif
 
 void z_sys_poweroff(void)
 {
 #ifdef CONFIG_STM32_WKUP_PINS
 	stm32_pwr_wkup_pin_cfg_pupd();
 #endif /* CONFIG_STM32_WKUP_PINS */
 
+#if DT_HAS_COMPAT_STATUS_OKAY(st_stm32u5_dma)
+	/* Turn off DMA for all UARTs to let the MCU to go into shutdown mode. */
```

Comment:
I think it would be nice to have a reference here to `Errata Sheet ES0499 : STM32U575xx and STM32U585xx` to workaround which uart_stm32 driver leaves DMA enable even upon completion of the transfers, hence the need to disable them here.